doc/devdocs/modules/FileExplorer/monaco/readme.md

```diff
@@ -0,0 +1,60 @@
+# Developer Preview (Monaco)
+
+Developer preview is based on [Microsoft's Monaco Editor](https://microsoft.github.io/monaco-editor/) which is maintained by the Visual Studio Code team.
+
+## Update Monaco Editor
+
+1. Download Monaco editor with [npm](https://www.npmjs.com/): `npm i monaco-editor`.
+2. Delete everything except the `min` folder (the minimised code) from the downloaded files.
+3. Copy the `min` folder inside the [`monacoSRC`](/src/modules/previewpane/MonacoPreviewHandler/monacoSRC) folder.
+4. Generate the JSON file as described in the [Generate monaco_languages.json file](#generate-monaco_languagesjson-file) section.
+
+## Add a new language definition
+
+1. Add the language definition (Written with [Monarch](https://microsoft.github.io/monaco-editor/monarch.html)) to the [`monacoSRC`](/src/modules/previewpane/MonacoPreviewHandler/customLanguages/) folder. The file should be formatted like this (change `idDefinition` to the name of your language):
+
+```javascript
+export function idDefinition() {
+    return {
+        ...
+    }
+}
+```
+
+2. Add the following line to the [`monacoSpecialLanguages.js` file](/src/modules/previewpane/MonacoPreviewHandler/monacoSpecialLanguages.js):
+
+```javascript
+import { idDefinition } from './customLanguages/file.js';
+```
+
+3. In the [`monacoSpecialLanguages.js` file](/src/modules/previewpane/MonacoPreviewHandler/monacoSpecialLanguages.js) add the following line to the `registerAdditionalLanguages` function:
+
+```javascript
+registerAdditionalNewLanguage("id", [".fileExtension"], idDefinition(), monaco)
+```
+
+4. Execute the steps described in the [Generate monaco_languages.json file](#generate-monaco_languagesjson-file) section.
+
+## Add a new file extension to an existing language
+
+1. In the [`monacoSpecialLanguages.js` file](/src/modules/previewpane/MonacoPreviewHandler/monacoSpecialLanguages.js) add the following line to the `registerAdditionalLanguages` function:
+
+```javascript
+registerAdditionalLanguage("oldIdExt", [".fileExtension"], "oldId", monaco)
```

Comment:
For me it is not clear what oldIdExt or oldId means. Maybe It would be if I look into the code. Can we add some examples here and would that make sense?